src/il_representations/scripts/interpret.py

```diff
@@ -371,42 +209,44 @@ def run(log_dir, chosen_algo, layer_kwargs, save_video, filename, dataset_config
                                          adjust_axis=False,
                                          make_grid=False)
 
-    for itr, (tensor_image, label) in enumerate(zip(images, labels)):
-        # Get policy prediction
-        tensor_image = tensor_image.contiguous()
+    for itr, (image, label) in enumerate(zip(images, labels)):
+        if itr > length:
+            break
+        tensor_image = torch.Tensor(image).unsqueeze(dim=0).to(device)
+        tensor_label = int(label)
+
+        # For continuous space actions, we don't need to provide a label since
+        # Captum assumes the provided label stands for "class_num" and is an
+        # integer.
+        # TODO: connect this with action space
+        # if len(tensor_label) > 1:
+        #     tensor_label = None
         interp_algo_func = interp_algos.get(chosen_algo)
 
-        if 'layer' in chosen_algo:
-            module, idx = layer_kwargs[chosen_algo]['module'], \
-                          layer_kwargs[chosen_algo]['layer_idx']
-            chosen_layer = choose_layer(network, module, idx)
-
-        interpreted_img = interp_algo_func(network, tensor_image, label)  # shape (600, 600, 3)
+        interpreted_img = interp_algo_func(network, tensor_image, tensor_label)
 
         if save_video:
             video_writer.add_tensor(preprocess_obs(interpreted_img,
                                                    observation_space,
                                                    normalize_images=True))
-        else:
-            save_img(interpreted_img,
-                     save_name=f'{chosen_algo}_{itr}',
-                     save_dir=log_dir,
-                     observation_space=observation_space)
+        if save_image:
+            Path(f'{log_dir}/images').mkdir(parents=True, exist_ok=True)
+            save_img(save_name=f'{filename}_{itr}',
+                     save_dir=f'{log_dir}/images')
+
+            if save_original_image:
+                processed_image = []
+                i = 0
+                while i < len(image):
+                    processed_image.append(1.0/3.0 * image[i:i+3].astype(float))
+                    i += 3
+                processed_image = np.sum(processed_image, axis=0).astype(int)
```

Comment:
```suggestion
                # convert to grayscale
                processed_image = np.mean(
                    image.reshape((3, -1) + image.shape[1:]), axis=0).astype(int)
```

I think (need to check) that this does the same thing as the original code, and it's a bit shorter/more straightforward.

Reply:
Also, meta-level point: I assumed this was converting to grayscale and discarding the channels axis, but then I'm a bit confused by the transpose line down the bottom, which seems to assume that axis 0 is still the channels axis. Could you please add a comment explaining what the data format of `image` is, and why this transformation makes sense?

Reply:
Okay, with the comment it makes sense. It's just averaging the three input frames right?

Reply:
Yes :)